test/jasmine/tests/polar_test.js

```diff
@@ -1034,7 +1073,226 @@ describe('Test polar interactions:', function() {
         .then(function() {
             expect(eventCnts.plotly_relayout).toBe(4, 'total # of relayout events');
         })
-        .catch(fail)
+        .catch(failTest)
+        .then(done);
+    });
+});
+
+describe('Test polar *gridshape linear* interactions', function() {
+    var gd;
+
+    beforeEach(function() {
+        jasmine.DEFAULT_TIMEOUT_INTERVAL = 5000;
+        gd = createGraphDiv();
+    });
+
+    afterEach(destroyGraphDiv);
+
+    it('should snap radial axis rotation to polygon vertex angles', function(done) {
+        var dragPos0 = [150, 25];
+        var dragPos1 = [316, 82];
+        var evtCnt = 0;
+
+        // use 'special' drag method - as we need two mousemove events
+        // to activate the radial drag mode
+        function _drag(p0, dp) {
+            var node = d3.select('.polar > .draglayer > .radialdrag').node();
+            return drag(node, dp[0], dp[1], null, p0[0], p0[1], 2);
+        }
+
+        function _assert(msg, angle) {
+            expect(gd._fullLayout.polar.radialaxis.angle)
+                .toBeCloseTo(angle, 1, msg + ' - angle');
+        }
+
+        Plotly.plot(gd, [{
+            type: 'scatterpolar',
+            // octogons have nice angles
+            r: [1, 2, 3, 2, 3, 1, 2, 1, 2],
+            theta: ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'a']
```

Comment:
hmm, not for this PR of course (and would be useful in circular mode as well), but this makes me realize we should have a mode in `scatterpolar` to join the endpoints together so you don't need to repeat the initial point when the data are meant to be periodic. `trace.connectends` perhaps, to mirror `trace.connectgaps`?

Reply:
Good call. Added to -> https://github.com/plotly/plotly.js/issues/2255